plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -811,6 +811,10 @@ def save_fileless_units(self, file_handle, tag, process_func, mutable_type=False
                 (model for model in all_packages if model.unit_key_as_named_tuple in to_save)
 
         for model in package_info_generator:
+            # Add repo_id to each collection of the pkglist of the new erratum
+            if isinstance(model, models.Errata):
+                for collection in model.pkglist:
+                    collection['_pulp_repo_id'] = self.repo.repo_id
```

Comment:
@bmbouter,
_pulp_repo_id is added to the collections to be able to handle errata with the same errata_id and with the same collection name but with the different packages inside this collection. E.g. if the same erratum was pushed for RHEL6 and RHEL7, in both repositories it will be exactly the same but the packages will differ.
Adding _pulp_repo_id is a workaround, because we are not able to modify unit_key (which is errata_id only) without making a backward incompatible change.
Here is an example of erratum and its pkglist https://paste.fedoraproject.org/360775/18653391/
Does it answer your question?